src/com/google/enterprise/adaptor/database/DatabaseAdaptor.java

```diff
@@ -378,14 +378,23 @@ private void addMetadata(MetadataHandler meta, ResultSet rs)
             }
           }
           break;
+        case Types.DATE:
+          Date dt = rs.getDate(index);
+          value = dt.toString();
```

Comment:
Handle null, here and below, and add tests for the same (for each data type, since you insist on forgetting it so often).